src/wallet/scriptpubkeyman.cpp

```diff
@@ -1552,12 +1552,61 @@ isminetype DescriptorScriptPubKeyMan::IsMine(const CScript& script) const
 
 bool DescriptorScriptPubKeyMan::CheckDecryptionKey(const CKeyingMaterial& master_key, bool accept_no_keys)
 {
-    return false;
+    {
+        LOCK(cs_desc_man);
+        if (!m_map_keys.empty())
+            return false;
+
+        bool keyPass = m_map_crypted_keys.empty(); // Always pass when there are no encrypted keys
+        bool keyFail = false;
+        CryptedKeyMap::const_iterator mi = m_map_crypted_keys.begin();
+        for (; mi != m_map_crypted_keys.end(); ++mi)
```

Comment:
`for (const auto& mi : m_map_crypted_keys) {` and then below `mi.second.first;` compiles too, though maybe I missed something.

Reply:
Done

Reply:
No, it seems like I've accidentally applied that change to `LegacyScriptPubKeyMan`. Applied these changes to the right place.